src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2372,7 +2372,10 @@ private FlattenTypeResult FlattenTypeHelper(MetadataType type, uint baseOffs, CO
         {
             if (type.IsExplicitLayout || (type.IsSequentialLayout && type.GetClassLayout().Size != 0) || type.IsInlineArray)
             {
-                significantPadding = true;
+                if (!type.ContainsGCPointers && !type.IsByRefLike)
+                {
+                    significantPadding = true;
```

Comment:
Does this pessimize cases that don't have padding?

Say for example you've defined a `union { float f; int32_t i; }` and therefore it is explicit, but the two structures are "perfectly overlapping".

-- JW as a potential "future" improvement, not something to be handled in this PR or anything

Reply:
No, since we don't support promotion of structs with overlapping fields. But if we did still no as the JIT only pays attention to `significantPadding` when there are holes in the struct.